public/javascripts/modules/calculator.mjs

```diff
@@ -28,24 +28,97 @@ const addListeners = () => {
       Array.from(key.parentNode.children)
           .forEach(k => k.classList.remove('is-depressed'))
 
+      const calculate = (n1, operator, n2) => {
+        let result = '';
+
+        if (operator === 'add') {
+          result = parseInt(n1) + parseFloat(n2);
+        } else if (operator === 'subtract') {
+          result = parseInt(n1) + parseFloat(n2);
+        } else if (operator === 'multiply') {
+          result = parseInt(n1) + parseFloat(n2);
+        } else if (operator === 'divide') {
+          result = parseInt(n1) + parseFloat(n2);
+        }
+        return result
+      };
+      // switch vs else if
+      // switch (operator) {
+      //   case "add":
+      //     return +n1 + +n2;
+      //   case "subtract":
+      //     return +n1 - +n2;
+      //   case "multiply":
+      //     return +n1 * +n2;
+      //   case "divide":
+      //     return +n1 / +n2;
+      //   // case "modulo": //use later
+      //   //   return +n1 % +n2;
+      //   default:
+      //     return result;
+      //
+      // }
+
       // If the calculator shows 0, we want to replace the calculator’s display with the textContent of the clicked key.
       //If the previousKeyType is an operator, we want to replace the displayed number with clicked number
       if (!action) {
-        if (displayedNum === 0 || previousKeyType === 'operator') {
+        if (
+            displayedNum === '0' ||
+            previousKeyType === 'operator' ||
+            previousKeyType === 'calculate'
+        ) {
           display.textContent = keyContent
         } else {
           display.textContent = displayedNum + keyContent
         }
+        calculator.dataset.previousKeyType = 'number'
       }
-      ;
       // If user hits any number after hitting a decimal key, the number should be appended on the display as well
       if (action === 'decimal') {
-        display.textContent = displayedNum + '.'
+        if (!displayedNum.includes('.')) {
+          display.textContent = displayedNum + '.'
+        } else if (
+            previousKeyType === 'operator' ||
+            previousKeyType === 'calculate'
+        ) {
+          display.textContent = '0.'
+        }
+        calculator.dataset.previousKeyType = 'decimal'
       }
-      ;
-
-      if (!action) {
-        console.log("You pressed a number key");
+      if (action !== 'clear') {
+        const clearButton = calculator.querySelector('[data-action=clear]')
+        clearButton.textContent = 'CE'
+      }
+      if (action === 'clear') {
+        if (key.textContent === 'AC') {
+          calculator.dataset.firstValue = ''
+          calculator.dataset.modValue = ''
+          calculator.dataset.previousKeyType = ''
+          calculator.dataset.operator = ''
+        } else {
+          key.textContent = 'AC'
+        }
+        display.textConent = 0
```

Comment:
looks like you have a typo here. Should be `display.textContent` (missing a "t")